dspace-api/src/main/java/org/dspace/app/util/SubmissionStepConfig.java

```diff
@@ -42,28 +48,31 @@
      * &lt;step-definitions&gt; section)
      */
     private String id = null;
+    
+    private boolean mandatory = true;
 
     /** the heading for this step */
     private String heading = null;
 
     /** the name of the java processing class for this step */
     private String processingClassName = null;
-   
-    /** whether or not this step is editable during workflow (default=true) */
-    private boolean workflowEditable = true;
 
     /** 
-     * The full name of the JSP-UI binding class for this step. This field is
-     * ONLY used by the JSP-UI.
+     * The name of the UI components for this step.
      **/
-    private String jspBindingClassName = null;
+    private String type = null;
+    
 
-    /**
-     * The full name of the Manakin XML-UI Transformer class which will generate
-     * the necessary DRI for displaying this class in Manakin. This field is
-     * ONLY used by the Manakin XML-UI.
-     */
-    private String xmlBindingClassName = null;
+    /** 
+     * The scope restriction for this step (submission or workflow).
+     **/
+    private String scope = null;
```

Comment:
To make the meaning of scope more clear. Can you change this to an enum with values `SUBMISSION`, `WORKFLOW`, `VERSIONING` ?

Reply:
IMHO it is early to introduce enumeration. Please see the contract how address the scope attribute https://github.com/DSpace/Rest7Contract/blob/master/submissionforms.md

Reply:
But since you have an [enum for this in the REST model](https://github.com/4Science/DSpace/blob/f755f59d381b5c166998318be02a81a456a5b560/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/ScopeEnum.java), wouldn't it make sense to also have a similar enum in the domain model?